pkg/util/http/client.go

```diff
@@ -31,86 +27,68 @@ func SanitizeURL(message string) string {
 	return apiKeyRegExp.ReplaceAllString(message, apiKeyReplacement)
 }
 
-// CreateHTTPTransport creates an *http.Transport for use in the agent
-func CreateHTTPTransport() *http.Transport {
-	tlsConfig := &tls.Config{
-		InsecureSkipVerify: config.Datadog.GetBool("skip_ssl_validation"),
+// Client wraps (http.Client).Do and resets the underlying connections at the
+// configured interval
+type Client struct {
```

Comment:
I would call it `ResetClient` because resetting is its main characteristic. This would also make it much easier to understand out of context. Even when reading the constructor `NewResetClient` the first argument makes sense.

Reply:
very good point, I'll change this

Reply:
+1

Reply:
(addressed)